crm_phonecall_timesheet/models/crm_lead.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Alexandre Díaz <alexandre.diaz@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+
+    @api.multi
+    def write(self, vals):
+        res = super().write(vals)
+        if 'project_id' in vals:
```

Comment:
You don't need all this code. Take into account that this is for timesheets in phone calls, not in leads.